pkg/tsdb/elasticsearch/elasticsearch.go

```diff
@@ -36,15 +36,21 @@ func ProvideService(httpClientProvider httpclient.Provider) *Service {
 }
 
 func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-	if len(req.Queries) == 0 {
-		return &backend.QueryDataResponse{}, fmt.Errorf("query contains no queries")
-	}
-
 	dsInfo, err := s.getDSInfo(req.PluginContext)
 	if err != nil {
 		return &backend.QueryDataResponse{}, err
 	}
 
+	// Support for version after their end-of-life (currently <7.10.0) was removed
+	lastSupportedVersion, _ := semver.NewVersion("7.10.0")
+	if dsInfo.ESVersion.LessThan(lastSupportedVersion) {
+		return &backend.QueryDataResponse{}, fmt.Errorf("support for elasticsearch versions after their end-of-life (currently versions < 7.10) was removed")
+	}
+
+	if len(req.Queries) == 0 {
+		return &backend.QueryDataResponse{}, fmt.Errorf("query contains no queries")
```

Comment:
We can probably improve this message in the future! (it's been there for a while and is not very remedial)

Reply:
hehe, there's a certain charm to the recursive nature of "query contains no queries" 😁